### PR TITLE
added template: qlik-sense-enterprise

### PR DESCRIPTION
**Pull request by clintcarr**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Changed download of install to use the customscriptextension
* Updated errors in September 2017 Patch 1 install
* Removed parameters from parameter file where a default value exists

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@clintcarr can you check the ci failures?  also check the BP guide for handling artifacts, looks like the problem might be the param value, instead of a defaultValue

**Comment by clintcarr** [comment]
Hi,  yes, understood.  I have made the changes, however it is failing for other reasons now.  The CI now states: No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

File: qlik-sense-enterprise/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+          "location": "[resourceGroup().location]",
+          "apiVersion": "2016-03-30",
+          "dependsOn": [
+            "[resourceId('Microsoft.Compute/virtualMachines', variables('vmName'))]"
+          ],
+          "tags": {
+            "displayName": "QlikSense"
+          },
+          "properties": {
+            "publisher": "Microsoft.Compute",
```
Comment (bmoore-msft) [inline qlik-sense-enterprise/azuredeploy.json]: Let's update this to 2.0...

            "publisher": "Microsoft.Azure.Extensions",
            "type": "CustomScript",
            "typeHandlerVersion": "2.0",

File: qlik-sense-enterprise/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "virtualMachineName": {
```
Comment (bmoore-msft) [inline qlik-sense-enterprise/azuredeploy.parameters.json]: for the hard-coded values here, just use defaultVaules in the template - keep just the GEN values.

File: qlik-sense-enterprise/scripts/qs-bootstrap.ps1 [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# QsBootstrap.ps1
+#
+$adminUser = $Args[0]
+$adminPassword = $Args[1]
+$scriptUrl = $($Args[10])
+$script = $($scriptUrl)+'/scripts/qs-install.ps1'
+$password =  ConvertTo-SecureString $($adminPassword) -AsPlainText -Force
+$credential = New-Object System.Management.Automation.PSCredential -ArgumentList $env:computername\$adminUser, $password
+New-Item -ItemType directory -Path C:\installation
+(New-Object System.Net.WebClient).DownloadFile($($script), "c:\installation\qs-install.ps1")
```
Comment (bmoore-msft) [inline qlik-sense-enterprise/scripts/qs-bootstrap.ps1]: use the custom script extension to download any script files needed

**Comment by clintcarr** [comment]
Changes requested made except for the API version.  Upon changing as per request the following occured:
{ "status": "Failed", "error": { "code": "ResourceDeploymentFailure", "message": "The resource operation completed with terminal provisioning state 'Failed'.", "details": [ { "code": "VMExtensionHandlerNonTransientError", "message": "Handler 'Microsoft.Azure.Extensions.CustomScript' has reported failure for VM Extension 'QlikSense' with terminal error code '1007' and error message: 'Install failed for plugin (name: Microsoft.Azure.Extensions.CustomScript, version 2.0.5) with exception The specified executable is not a valid application for this OS platform.'" } ] } }

Using the Microsoft documentation (https://docs.microsoft.com/en-us/azure/virtual-machines/windows/classic/extensions-customscript) I updated from 1.4 to 1.8.

**Pull request merged**